### DISCUSSION TITLE
Query returning old data

### DISCUSSION THREAD

**Discussion by aarjan** [prisma/prisma#18401] [Q&A]
I used prisma.queryRaw in my express app, but returns stale data. 
I updated the data from the mysql console, but querying the data through prisma.
I am not sure if prisma caches the results or not as i couldn't find anything related to it in the docs.

I get the accurate results while querying in the mysql console, but the output of the prisma query is stale.
It seems to work after some time like 5mins, or if i restart the process.

I ran the same prisma raw query from a simple executable TS script, it would return the latest results.

**Comment by ludralph** [comment]
Hi @aarjan 👋 

Prisma queryRaw doesn't cache data as it is executed against the DB. Can you share the relevant part of the code so we can examine it?

**Reply by nurul3101** [reply]
To add, What query is being invoked in queryRaw? Is it a transaction? Open transactions can lock the data and prevent it from being updated. If you have any open transactions, that might cause this issue.

**Reply by aarjan** [reply]
It's a simple query something like, 

schema:
table: students
column: created_at int -> that stores unix_timestamps

query1:
``` {"message":"{\n  message: {\n    timestamp: 2023-03-21T12:25:26.619Z,\n    query: \"SELECT COUNT(*) FROM students WHERE unix_timestamp() > created_at",\n    params: '[]',\n    duration: 2,\n    target: 'quaint::connector::metrics'\n  },\n  level: 'debug',\n  timestamp: '2023-03-21 12:25:26'\n}\n","timestamp":"2023-03-21T12:25:26.619Z","type":"out","process_id":0,"app_name":"apiserver"}```


and then I did this from mysql console, 
update query2:
```update students set created_at= unix_timestamp(); ```

The prisma query would return 0 count whereas, the mysql console would return accurate results.

I didn't use any transactions.

**Reply by nurul3101** [reply]
Are you using the latest prisma version?

**Reply by nurul3101** [reply]
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
